File: cpp/2286.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    int count = 1, n, a, b;
    string even, odd;
    while (cin >> n) {
        if (n == 0) {
            break;
        }
        cin >> even >> odd;
        cout << "Teste " << count << '\n';
        for (int i = 0; i < n; i++) {
            cin >> a >> b;
            if ((a + b) % 2 == 0) {
                cout << even << '\n';
            }
            else {
                cout << odd << '\n';
            }
        }
        cout << '\n';
        count++;
    }
    return 0;
}

File: py/2286.py
def main() -> None:
    count = 1
    while True:
        n = int(input())
        if n == 0:
            break
        even, odd = input(), input()
        print('Teste', count)
        for _ in range(n):
            result = sum([int(i) for i in input().split()])
            if result % 2 == 0:
                print(even)
            else:
                print(odd)
        print('')
        count += 1


if __name__ == '__main__':
    main()
